pkg/cloudprovider/types.go

```diff
@@ -59,25 +59,28 @@ type NodeGroup interface {
 type Capacity interface {
 	// Create a set of nodes to fulfill the desired capacity given constraints.
 	Create(context.Context, *Constraints) ([]Packing, error)
-
-	// GetTopologyDomains returns a list of topology domains supported by the
-	// cloud provider for the given key.
-	// For example, GetTopologyDomains("zone") -> [ "us-west-2a", "us-west-2b" ]
-	// This enables the caller to to build Constraints for a known set of
-	GetTopologyDomains(context.Context, TopologyKey) ([]string, error)
+	// GetZones returns a list of zones supported by the cloud provider.
+	GetZones(context.Context) ([]string, error)
 }
 
-// Constraints lets the controller define the desired capacity,
-// avalability zone, architecture for the desired nodes.
+// Constraints for an efficient binpacking solution of pods onto nodes, given
+// overhead and node constraints.
 type Constraints struct {
-	// Pods is a list of equivalently schedulable pods to be efficiently
-	// binpacked.
+	NodeConstraints
+	// Pods is a list of equivalently schedulable pods to be binpacked.
 	Pods []*v1.Pod
-	// Overhead resources per node from system resources such a kubelet and
-	// daemonsets.
+	// Overhead resources per node from system resources such as kubelet and daemonsets.
 	Overhead v1.ResourceList
```

Comment:
What about referencing capacity or allocatable as opposed to Overhead? 
It seems more intuitive to me to reference the resources that are left in the context of constraints.

Reply:
I don't know the capacity or allocatable, since I don't know the ceiling. I do know how much overhead of the total node I need, though.